### DISCUSSION TITLE
Interactable Object - Collision

### DISCUSSION THREAD

**Discussion by DrDallas1** [SmartlyDressedGames/Unturned-3.x-Community#4051] [Ideas]
Hello! 

I have spoken with a few well known NPC modders in the community (namely Loco) extensively about a new addition that would benefit the modding scene a lot. Interactable collision quest objects.

What are they? Objects that would grant you NPC rewards when you collide with them (as opposed to pressing F on them)

What is the format?
Interactable Collision in the .dat file. Interactibility condition would determine if the collision occurs or not. “Conditions” would still determine if the object appears or not, which would override Interactibility condition (must pass both) 

It is almost a necessity that Interactibility_Effect should also be transferred and work on collision, and be transferred client side.

Unity:
A gameobject “Collision” would be used. All child colliders under that gameobject that have isTrigger set to true would trigger the collision.

Why?
Allows for scripted cutscenes, set pieces, and general almost scripting of new game modes. If used in a very clever way, could even create entire game modes that require checks for collision. 

Please do consider adding this. It would greatly benefit many.

**Comment by AMOXVaclav** [comment]
The posibilities would be endless and it would give the NPC modding scene so much new to work with. Currently there's an issue on Multiplayer servers where if you hide an object using a flag, players still cannot go through that place even though for them the object isn't there. With Interactable Collision, this problem would be completely solved.

**Comment by Jdance-Media** [comment]
~~I'd like to note that is currently is possible, but it requires us to craft up our own modhooks using `InteractableObjectQuest`.~~
We are currently testing this WIP theory.

**Comment by Kopfstroh** [comment]
I dont understand, explain again please

**Comment by DrDallas1** [comment]
> I'd like to note that is currently is possible, but it requires us to craft up our own modhooks using `InteractableObjectQuest`.

For info, we are currently investigating this with Loco, Gamingtoday, Pixel and you Jdance. Currently no luck.

**Comment by DrDallas1** [comment]
> I dont understand, explain again please

Basically, an object that where you collide with it, it will give you NPC rewards. 

Example: Collide with a trigger, and you would be given a reward. Enter a certain location, and you would be given specific items.

**Comment by Jdance-Media** [comment]
We did some digging, our idea to use `InteractableObjectQuest` did not work out as then there will be two different Interactable components. We think the idea for #4053 would likely be able to remove this roadblock for us and also save you (Nelson) work.

**Comment by SDGNelson** [comment]
Something on my mind to do has been an asset that can specify the NPC conditions/rewards, and then a volume could link this asset to apply when a player enters, so it should happen eventually ™
